From c8380d1bc4f2a3533fe199d72d5103697729ea58 Mon Sep 17 00:00:00 2001
From: zachary <zhangzg@marvell.com>
Date: Tue, 26 Jan 2016 17:32:23 +0800
Subject: [PATCH 0782/1240] fix: ddr: a3700: update DDR size to the value
 passed to u-boot by SPL

- DDR size has been read from dts DDR node in SPL by ddr driver,
  and pass to u-boot with set_info feature, which puts all sys_info
  struct in DRAM under address 0x4000000 as an array.
- In u-boot get the value and set it to gd->ram_size.

Change-Id: Ibd12dcb95108cfd73952a33a3005294f5c5c9f32
Signed-off-by: zachary <zhangzg@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/26240
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Wilson Ding <dingwei@marvell.com>
---
 arch/arm/cpu/armv8/armadalp/soc-init.c | 20 +++-----------------
 1 file changed, 3 insertions(+), 17 deletions(-)

diff --git a/arch/arm/cpu/armv8/armadalp/soc-init.c b/arch/arm/cpu/armv8/armadalp/soc-init.c
index e53c047..66ad0bf 100644
--- a/arch/arm/cpu/armv8/armadalp/soc-init.c
+++ b/arch/arm/cpu/armv8/armadalp/soc-init.c
@@ -133,25 +133,11 @@ int dram_init(void)
 	/* NO DRAM init sequence in Pallaidum, so set static DRAM size of 256MB */
 	gd->ram_size = 0x20000000;
 #else
-	struct mbus_win_map win_map;
-	struct mbus_win *win;
-	int cs;
-	u32 size;
-
 	gd->ram_size = 0;
 
-	/* get DRAM window configuration from MBUS */
-	mbus_win_map_build(&win_map);
-
-	for (cs = 0, win = &win_map.mbus_windows[cs]; cs < win_map.mbus_win_num; cs++, win++) {
-			gd->bd->bi_dram[cs].start = win->base_addr;
-			size = (win->win_size + 1) * MBUS_CR_WIN_SIZE_ALIGNMENT;
-			gd->bd->bi_dram[cs].size = size;
-
-			gd->ram_size += size;
-
-			debug("DRAM bank %d base 0x%08x size 0x%x ", cs, (u32)win->base_addr, size);
-	}
+	/* DDR size has been read from dts DDR node in SPL
+	 * ddr driver and pass to u-boot. */
+	gd->ram_size = (get_info(DRAM_CS0_SIZE) << 20);
 
 	if (gd->ram_size == 0) {
 		error("No DRAM banks detected");
-- 
1.9.1

